### PR TITLE
auth: move fmt::formatter<auth::resource_kind> up

### PR DESCRIPTION
**Pull request by tchaikov**
before this change, `fmt::formatter<auth::resource_kind>` is located at line 250 in this file, but it is used at line 130. so, {fmt} is not able to find it:

```
/usr/include/fmt/core.h:2593:45: error: implicit instantiation of undefined template 'fmt::detail::type_is_unformattable_for<auth::resource_kind, char>'
 2593 |     type_is_unformattable_for<T, char_type> _;
      |                                             ^
/usr/include/fmt/core.h:2656:23: note: in instantiation of function template specialization 'fmt::detail::parse_format_specs<auth::resource_kind, fmt::detail::compile_parse_context<char>>' requested here
 2656 |         parse_funcs_{&parse_format_specs<Args, parse_context_type>...} {}
      |                       ^
/usr/include/fmt/core.h:2787:47: note: in instantiation of member function 'fmt::detail::format_string_checker<char, auth::resource_kind, auth::resource_kind>::format_string_checker' requested here
 2787 |       detail::parse_format_string<true>(str_, checker(s));
      |                                               ^
/home/kefu/dev/scylladb/auth/resource.hh:130:29: note: in instantiation of function template specialization 'fmt::basic_format_string<char, auth::resource_kind &, auth::resource_kind &>::basic_format_string<char[65], 0>' requested here
  130 |             seastar::format("This resource has kind '{}', but was expected to have kind '{}'.", actual, expected)) {
      |                             ^
/usr/include/fmt/core.h:1578:45: note: template is declared here
 1578 | template <typename T, typename Char> struct type_is_unformattable_for;
      |                                             ^
```

in this change, `fmt::formatter<auth::resource_kind>` is moved up to where `auth::resource_kind` is defined. so that it can be used by its caller.

Refs #13245

### CODE REVIEW

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hello maintainers, could you help review this change?

**Comment by xemul** [comment]
How did the tree compile w/o this patch then?

**Comment by tchaikov** [comment]
> How did the tree compile w/o this patch then?

@xemul Hi Pavel, that's a good question. 

tl;dr: i don't have an explanation at this moment. and i haven't tried to find one.

long story: as you point out, the tree builds just fine in the dbuild env. but my environment is different than the building environment in our jenkins and probably some other developers. as i am using {fmt} 10.2.1, clang-19, and libstdc++-13, and i always use cmake when building the tree.  if i run into a build failure, i always try to understand it. if i manage to convince myself that it should also applies to upstream, i am tempted to send a patch. otherwise i just keep it in my own repo and wait for the big moment =)
